pkg/apis/trainer/v1alpha1/trainingruntime_types.go

```diff
@@ -173,7 +177,9 @@ type TorchMLPolicySource struct {
 	// Supported values: `auto`, `cpu`, `gpu`, or int value.
 	// TODO (andreyvelich): Add kubebuilder validation.
 	// Defaults to `auto`.
-	NumProcPerNode *string `json:"numProcPerNode,omitempty"`
+	// +kubebuilder:default="auto"
+	// +kubebuilder:validation:XValidation:rule="self in ['auto', 'cpu', 'gpu'] || type(self) == int", message="NumProcPerNode must be equal to auto, cpu, gpu, or int value"
+	NumProcPerNode *intstr.IntOrString `json:"numProcPerNode,omitempty"`
```

Comment:
@akshaychitneni @astefanutti @kubeflow/wg-training-leads @tenzen-y @Electronic-Waste In general, what are the benefits for the end-user to have `IntOrString` for `NumProcPerNode` API ?
E.g. we can always validate that this value is `int` if selected TrainingRuntime doesn't support `string` values.

Reply:
cc @kannon92 for feedback here.

Reply:
default is set to auto though?

Reply:
Not really, since `mpirun` doesn't support auto for OpenMPI:
https://www.open-mpi.org/doc/v4.1/man1/mpirun.1.php#toc9

E.g. in MPI slot == proc

Reply:
From the OpenAPI perspective do we see any problems to keep `intOrString` type ?

Reply:
Can we revert this API change in this PR and discuss it in the separate issue ?
So we can unblock other changes for TrainingRuntime validation in this PR.

Reply:
I fully agree with using `*intstr.IntOrString` better typed power.
@andreyvelich It seems that only you have objections for `intstr.IntOrString`, maybe?

Reply:
Actually, I asked about whether this type is ok to use from the API perspective (e.g. be compatible with OpenAPI and other formats).
For example, if someone is building API Server on top of TrainingRuntimes, it might be hard to implement `IntOrString` type since it is not a default type.

Reply:
@tenzen-y Don't you see any concerns with that ?
Does SIG Architecture in k8s have any recommendations for it ?
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties